erts/emulator/beam/copy.c

```diff
@@ -116,10 +117,14 @@ Uint size_object_x(Eterm obj, erts_literal_area_t *litopt)
 		case ARITYVAL_SUBTAG:
 		    ptr = tuple_val(obj);
 		    arity = header_arity(hdr);
-		    sum += arity + 1;
 		    if (arity == 0) { /* Empty tuple -- unusual. */
+                        /*
+                           The empty tuple is always a literal so
+                           it does not take up any extra space
+                         */
 			goto pop_next;
```

Comment:
Add `ASSERT(erts_is_literal(obj,ptr));`

and move down `ptr = tuple_val(obj);` after `if`.

Reply:
Thanks, fixed!